src/harness/fourslash.ts

```diff
@@ -245,12 +245,41 @@ namespace FourSlash {
             }
         }
 
+        static mockResolveModule(moduleName: string, _initialDir: string, _host: ts.server.ServerHost, log: (message: string) => void): ts.server.PluginModule {
+            log(`Mock resolving ${moduleName}`);
+            return {
+                create(info: ts.server.PluginCreateInfo) {
+                    // tslint:disable-next-line:no-null-keyword
+                    const proxy = Object.create(null);
+                    const langSvc: any = info.languageService;
+                    for (const k of Object.keys(langSvc)) {
+                        proxy[k] = function () {

```

Comment:
The LS codebase (and the API's consumers) are not resilient to exceptions being thrown from these API methods, however consumers of the plugin API are not bound by any contract forbidding exceptions. Is there some way we can force a safer construction of these proxy methods whereby execution errors caused by improper plugins are appropriately logged and recorded, rather than surfaced as ICEs?